bpf/lib/nodeport.h

```diff
@@ -104,11 +104,12 @@ static __always_inline int nodeport_snat_fwd_ipv6(struct __ctx_buff *ctx,
 		.min_port = NODEPORT_PORT_MIN_NAT,
 		.max_port = NODEPORT_PORT_MAX_NAT,
 	};
-	int ret = CTX_ACT_OK;
-	bool snat_needed;
+	int snat_needed, ret = CTX_ACT_OK;
 
 	snat_needed = snat_v6_prepare_state(ctx, &target);
-	if (snat_needed)
+	if (IS_ERR(snat_needed) && snat_needed != NAT_PUNT_TO_STACK)
+		return snat_needed;
+	if (snat_needed != NAT_PUNT_TO_STACK)
```

Comment:
I *think* the following should be equivalent, and hopefully easier to read:

```suggestion
	ret = snat_v6_prepare_state(ctx, &target);
	if (ret == NAT_NEEDED)
		ret = snat_v6_nat(ctx, &target, ext_err);

	if (ret == NAT_PUNT_TO_STACK)
		ret = CTX_ACT_OK;

	ctx_snat_done_set(ctx);
	return ret;
```

Reply:
Yes, I hesitated between keeping exactly the same checks or using the new `NAT_NEEDED`. I also believe this would be more readable, I'll go for that. Thanks!